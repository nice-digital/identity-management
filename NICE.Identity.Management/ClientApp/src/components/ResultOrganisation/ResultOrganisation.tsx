import React, { FC } from "react";
import { Link } from "react-router-dom";
import Moment from "moment";
import { Card } from "@nice-digital/nds-card";
import { type OrganisationType } from "src/models/types";

type CardMetaData = {
	label?: string;
	value: string;
};

type ResultsOrganisationProps = {
	item: OrganisationType;
};

export const ResultOrganisation: FC<ResultsOrganisationProps> = ({ item }) => {
	const { id, name, dateAdded } = item;
	const organisationsListHeading = {
		headingText: `${name}`,
		link: {
			elementType: Link,
			destination: `/organisations/${id}`,
		},
	};

	const organisationsListMetadata: Array<CardMetaData> = [
		{
			label: "Date created",
			value: Moment(dateAdded).format("D MMMM YYYY HH:mm"),
		},
	];

	return (
		<Card {...organisationsListHeading} metadata={organisationsListMetadata} />
	);
};
