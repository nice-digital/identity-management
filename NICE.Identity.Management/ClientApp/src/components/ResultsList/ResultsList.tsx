import React, { ElementType } from "react";

type ResultsListProps<T> = {
	data: T[];
	elementType: ElementType<{ item: T }>;
	qaSelExtract: string;
};

export const ResultsList = <T,>({
	data,
	elementType: ElementType,
	qaSelExtract,
}: ResultsListProps<T>): React.ReactElement => (
	<ul className="list--unstyled mt--d" aria-label={qaSelExtract} data-qa-sel={`list-of-${qaSelExtract}`}>
		{data.map((item, index) => (
			<li key={index}>
				<ElementType item={item} />
			</li>
		))}
	</ul>
);
