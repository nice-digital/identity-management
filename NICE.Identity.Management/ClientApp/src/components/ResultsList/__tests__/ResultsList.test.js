import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import { ResultsList } from "../ResultsList";
import { ResultOrganisation } from "../../ResultOrganisation/ResultOrganisation";
import organisations from "./organisations.json";

test("should match the snapshot when ResultOrganisation has been passed into elementType", async () => {
	const resultsListOrganisationProps = {
		data: organisations,
		elementType: ResultOrganisation,
		qaSelExtract: "organisations",
	};
	const {container} =	render(<ResultsList {...resultsListOrganisationProps} />, {wrapper: MemoryRouter});
	expect(container).toMatchSnapshot();
});