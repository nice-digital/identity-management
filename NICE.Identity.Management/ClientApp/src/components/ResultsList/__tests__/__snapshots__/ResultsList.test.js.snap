// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot when ResultOrganisation has been passed into elementType 1`] = `
<div>
  <ul
    aria-label="organisations"
    class="list--unstyled"
    data-qa-sel="list-of-organisations"
  >
    <li>
      <article
        class="card"
      >
        <header
          class="card__header"
        >
          <p
            class="card__heading"
          >
            <a
              href="/organisations/1001"
            >
              NICE
            </a>
          </p>
        </header>
        <dl
          class="card__metadata"
        >
          <div
            class="card__metadatum"
          >
            <dt
              class="visually-hidden"
            >
              Date created
            </dt>
            <dd>
              1 December 2021 00:00
            </dd>
          </div>
        </dl>
      </article>
    </li>
    <li>
      <article
        class="card"
      >
        <header
          class="card__header"
        >
          <p
            class="card__heading"
          >
            <a
              href="/organisations/1003"
            >
              Org 1
            </a>
          </p>
        </header>
        <dl
          class="card__metadata"
        >
          <div
            class="card__metadatum"
          >
            <dt
              class="visually-hidden"
            >
              Date created
            </dt>
            <dd>
              11 January 2022 13:13
            </dd>
          </div>
        </dl>
      </article>
    </li>
    <li>
      <article
        class="card"
      >
        <header
          class="card__header"
        >
          <p
            class="card__heading"
          >
            <a
              href="/organisations/1004"
            >
              Org 2
            </a>
          </p>
        </header>
        <dl
          class="card__metadata"
        >
          <div
            class="card__metadatum"
          >
            <dt
              class="visually-hidden"
            >
              Date created
            </dt>
            <dd>
              12 January 2022 13:14
            </dd>
          </div>
        </dl>
      </article>
    </li>
    <li>
      <article
        class="card"
      >
        <header
          class="card__header"
        >
          <p
            class="card__heading"
          >
            <a
              href="/organisations/1005"
            >
              Org 3
            </a>
          </p>
        </header>
        <dl
          class="card__metadata"
        >
          <div
            class="card__metadatum"
          >
            <dt
              class="visually-hidden"
            >
              Date created
            </dt>
            <dd>
              13 January 2022 13:15
            </dd>
          </div>
        </dl>
      </article>
    </li>
    <li>
      <article
        class="card"
      >
        <header
          class="card__header"
        >
          <p
            class="card__heading"
          >
            <a
              href="/organisations/1006"
            >
              Org 4
            </a>
          </p>
        </header>
        <dl
          class="card__metadata"
        >
          <div
            class="card__metadatum"
          >
            <dt
              class="visually-hidden"
            >
              Date created
            </dt>
            <dd>
              1 December 2021 00:00
            </dd>
          </div>
        </dl>
      </article>
    </li>
    <li>
      <article
        class="card"
      >
        <header
          class="card__header"
        >
          <p
            class="card__heading"
          >
            <a
              href="/organisations/1007"
            >
              Org 5
            </a>
          </p>
        </header>
        <dl
          class="card__metadata"
        >
          <div
            class="card__metadatum"
          >
            <dt
              class="visually-hidden"
            >
              Date created
            </dt>
            <dd>
              11 January 2022 13:13
            </dd>
          </div>
        </dl>
      </article>
    </li>
    <li>
      <article
        class="card"
      >
        <header
          class="card__header"
        >
          <p
            class="card__heading"
          >
            <a
              href="/organisations/1008"
            >
              Org 6
            </a>
          </p>
        </header>
        <dl
          class="card__metadata"
        >
          <div
            class="card__metadatum"
          >
            <dt
              class="visually-hidden"
            >
              Date created
            </dt>
            <dd>
              12 January 2022 13:14
            </dd>
          </div>
        </dl>
      </article>
    </li>
    <li>
      <article
        class="card"
      >
        <header
          class="card__header"
        >
          <p
            class="card__heading"
          >
            <a
              href="/organisations/1009"
            >
              Org 7
            </a>
          </p>
        </header>
        <dl
          class="card__metadata"
        >
          <div
            class="card__metadatum"
          >
            <dt
              class="visually-hidden"
            >
              Date created
            </dt>
            <dd>
              13 January 2022 13:15
            </dd>
          </div>
        </dl>
      </article>
    </li>
    <li>
      <article
        class="card"
      >
        <header
          class="card__header"
        >
          <p
            class="card__heading"
          >
            <a
              href="/organisations/1010"
            >
              Org 8
            </a>
          </p>
        </header>
        <dl
          class="card__metadata"
        >
          <div
            class="card__metadatum"
          >
            <dt
              class="visually-hidden"
            >
              Date created
            </dt>
            <dd>
              1 December 2021 00:00
            </dd>
          </div>
        </dl>
      </article>
    </li>
    <li>
      <article
        class="card"
      >
        <header
          class="card__header"
        >
          <p
            class="card__heading"
          >
            <a
              href="/organisations/1011"
            >
              Org 9
            </a>
          </p>
        </header>
        <dl
          class="card__metadata"
        >
          <div
            class="card__metadatum"
          >
            <dt
              class="visually-hidden"
            >
              Date created
            </dt>
            <dd>
              11 January 2022 13:13
            </dd>
          </div>
        </dl>
      </article>
    </li>
    <li>
      <article
        class="card"
      >
        <header
          class="card__header"
        >
          <p
            class="card__heading"
          >
            <a
              href="/organisations/1012"
            >
              Org 10
            </a>
          </p>
        </header>
        <dl
          class="card__metadata"
        >
          <div
            class="card__metadatum"
          >
            <dt
              class="visually-hidden"
            >
              Date created
            </dt>
            <dd>
              12 January 2022 13:14
            </dd>
          </div>
        </dl>
      </article>
    </li>
    <li>
      <article
        class="card"
      >
        <header
          class="card__header"
        >
          <p
            class="card__heading"
          >
            <a
              href="/organisations/1013"
            >
              Org 11
            </a>
          </p>
        </header>
        <dl
          class="card__metadata"
        >
          <div
            class="card__metadatum"
          >
            <dt
              class="visually-hidden"
            >
              Date created
            </dt>
            <dd>
              13 January 2022 13:15
            </dd>
          </div>
        </dl>
      </article>
    </li>
    <li>
      <article
        class="card"
      >
        <header
          class="card__header"
        >
          <p
            class="card__heading"
          >
            <a
              href="/organisations/1014"
            >
              Org 12
            </a>
          </p>
        </header>
        <dl
          class="card__metadata"
        >
          <div
            class="card__metadatum"
          >
            <dt
              class="visually-hidden"
            >
              Date created
            </dt>
            <dd>
              1 December 2021 00:00
            </dd>
          </div>
        </dl>
      </article>
    </li>
    <li>
      <article
        class="card"
      >
        <header
          class="card__header"
        >
          <p
            class="card__heading"
          >
            <a
              href="/organisations/1015"
            >
              Org 13
            </a>
          </p>
        </header>
        <dl
          class="card__metadata"
        >
          <div
            class="card__metadatum"
          >
            <dt
              class="visually-hidden"
            >
              Date created
            </dt>
            <dd>
              11 January 2022 13:13
            </dd>
          </div>
        </dl>
      </article>
    </li>
    <li>
      <article
        class="card"
      >
        <header
          class="card__header"
        >
          <p
            class="card__heading"
          >
            <a
              href="/organisations/1016"
            >
              Org 14
            </a>
          </p>
        </header>
        <dl
          class="card__metadata"
        >
          <div
            class="card__metadatum"
          >
            <dt
              class="visually-hidden"
            >
              Date created
            </dt>
            <dd>
              12 January 2022 13:14
            </dd>
          </div>
        </dl>
      </article>
    </li>
    <li>
      <article
        class="card"
      >
        <header
          class="card__header"
        >
          <p
            class="card__heading"
          >
            <a
              href="/organisations/1017"
            >
              Org 15
            </a>
          </p>
        </header>
        <dl
          class="card__metadata"
        >
          <div
            class="card__metadatum"
          >
            <dt
              class="visually-hidden"
            >
              Date created
            </dt>
            <dd>
              13 January 2022 13:15
            </dd>
          </div>
        </dl>
      </article>
    </li>
    <li>
      <article
        class="card"
      >
        <header
          class="card__header"
        >
          <p
            class="card__heading"
          >
            <a
              href="/organisations/1018"
            >
              Org 16
            </a>
          </p>
        </header>
        <dl
          class="card__metadata"
        >
          <div
            class="card__metadatum"
          >
            <dt
              class="visually-hidden"
            >
              Date created
            </dt>
            <dd>
              1 December 2021 00:00
            </dd>
          </div>
        </dl>
      </article>
    </li>
    <li>
      <article
        class="card"
      >
        <header
          class="card__header"
        >
          <p
            class="card__heading"
          >
            <a
              href="/organisations/1019"
            >
              Org 17
            </a>
          </p>
        </header>
        <dl
          class="card__metadata"
        >
          <div
            class="card__metadatum"
          >
            <dt
              class="visually-hidden"
            >
              Date created
            </dt>
            <dd>
              11 January 2022 13:13
            </dd>
          </div>
        </dl>
      </article>
    </li>
    <li>
      <article
        class="card"
      >
        <header
          class="card__header"
        >
          <p
            class="card__heading"
          >
            <a
              href="/organisations/1020"
            >
              Org 18
            </a>
          </p>
        </header>
        <dl
          class="card__metadata"
        >
          <div
            class="card__metadatum"
          >
            <dt
              class="visually-hidden"
            >
              Date created
            </dt>
            <dd>
              12 January 2022 13:14
            </dd>
          </div>
        </dl>
      </article>
    </li>
    <li>
      <article
        class="card"
      >
        <header
          class="card__header"
        >
          <p
            class="card__heading"
          >
            <a
              href="/organisations/1021"
            >
              Org 19
            </a>
          </p>
        </header>
        <dl
          class="card__metadata"
        >
          <div
            class="card__metadatum"
          >
            <dt
              class="visually-hidden"
            >
              Date created
            </dt>
            <dd>
              13 January 2022 13:15
            </dd>
          </div>
        </dl>
      </article>
    </li>
    <li>
      <article
        class="card"
      >
        <header
          class="card__header"
        >
          <p
            class="card__heading"
          >
            <a
              href="/organisations/1022"
            >
              Org 20
            </a>
          </p>
        </header>
        <dl
          class="card__metadata"
        >
          <div
            class="card__metadatum"
          >
            <dt
              class="visually-hidden"
            >
              Date created
            </dt>
            <dd>
              1 December 2021 00:00
            </dd>
          </div>
        </dl>
      </article>
    </li>
    <li>
      <article
        class="card"
      >
        <header
          class="card__header"
        >
          <p
            class="card__heading"
          >
            <a
              href="/organisations/1023"
            >
              Org 21
            </a>
          </p>
        </header>
        <dl
          class="card__metadata"
        >
          <div
            class="card__metadatum"
          >
            <dt
              class="visually-hidden"
            >
              Date created
            </dt>
            <dd>
              11 January 2022 13:13
            </dd>
          </div>
        </dl>
      </article>
    </li>
    <li>
      <article
        class="card"
      >
        <header
          class="card__header"
        >
          <p
            class="card__heading"
          >
            <a
              href="/organisations/1024"
            >
              Org 22
            </a>
          </p>
        </header>
        <dl
          class="card__metadata"
        >
          <div
            class="card__metadatum"
          >
            <dt
              class="visually-hidden"
            >
              Date created
            </dt>
            <dd>
              12 January 2022 13:14
            </dd>
          </div>
        </dl>
      </article>
    </li>
    <li>
      <article
        class="card"
      >
        <header
          class="card__header"
        >
          <p
            class="card__heading"
          >
            <a
              href="/organisations/1025"
            >
              Org 23
            </a>
          </p>
        </header>
        <dl
          class="card__metadata"
        >
          <div
            class="card__metadatum"
          >
            <dt
              class="visually-hidden"
            >
              Date created
            </dt>
            <dd>
              13 January 2022 13:15
            </dd>
          </div>
        </dl>
      </article>
    </li>
    <li>
      <article
        class="card"
      >
        <header
          class="card__header"
        >
          <p
            class="card__heading"
          >
            <a
              href="/organisations/1026"
            >
              Org 24
            </a>
          </p>
        </header>
        <dl
          class="card__metadata"
        >
          <div
            class="card__metadatum"
          >
            <dt
              class="visually-hidden"
            >
              Date created
            </dt>
            <dd>
              1 December 2021 00:00
            </dd>
          </div>
        </dl>
      </article>
    </li>
    <li>
      <article
        class="card"
      >
        <header
          class="card__header"
        >
          <p
            class="card__heading"
          >
            <a
              href="/organisations/1027"
            >
              Org 25
            </a>
          </p>
        </header>
        <dl
          class="card__metadata"
        >
          <div
            class="card__metadatum"
          >
            <dt
              class="visually-hidden"
            >
              Date created
            </dt>
            <dd>
              11 January 2022 13:13
            </dd>
          </div>
        </dl>
      </article>
    </li>
    <li>
      <article
        class="card"
      >
        <header
          class="card__header"
        >
          <p
            class="card__heading"
          >
            <a
              href="/organisations/1028"
            >
              Org 26
            </a>
          </p>
        </header>
        <dl
          class="card__metadata"
        >
          <div
            class="card__metadatum"
          >
            <dt
              class="visually-hidden"
            >
              Date created
            </dt>
            <dd>
              12 January 2022 13:14
            </dd>
          </div>
        </dl>
      </article>
    </li>
    <li>
      <article
        class="card"
      >
        <header
          class="card__header"
        >
          <p
            class="card__heading"
          >
            <a
              href="/organisations/1029"
            >
              Org 27
            </a>
          </p>
        </header>
        <dl
          class="card__metadata"
        >
          <div
            class="card__metadatum"
          >
            <dt
              class="visually-hidden"
            >
              Date created
            </dt>
            <dd>
              13 January 2022 13:15
            </dd>
          </div>
        </dl>
      </article>
    </li>
    <li>
      <article
        class="card"
      >
        <header
          class="card__header"
        >
          <p
            class="card__heading"
          >
            <a
              href="/organisations/1030"
            >
              Org 28
            </a>
          </p>
        </header>
        <dl
          class="card__metadata"
        >
          <div
            class="card__metadatum"
          >
            <dt
              class="visually-hidden"
            >
              Date created
            </dt>
            <dd>
              1 December 2021 00:00
            </dd>
          </div>
        </dl>
      </article>
    </li>
    <li>
      <article
        class="card"
      >
        <header
          class="card__header"
        >
          <p
            class="card__heading"
          >
            <a
              href="/organisations/1031"
            >
              Org 29
            </a>
          </p>
        </header>
        <dl
          class="card__metadata"
        >
          <div
            class="card__metadatum"
          >
            <dt
              class="visually-hidden"
            >
              Date created
            </dt>
            <dd>
              11 January 2022 13:13
            </dd>
          </div>
        </dl>
      </article>
    </li>
    <li>
      <article
        class="card"
      >
        <header
          class="card__header"
        >
          <p
            class="card__heading"
          >
            <a
              href="/organisations/1032"
            >
              Org 30
            </a>
          </p>
        </header>
        <dl
          class="card__metadata"
        >
          <div
            class="card__metadatum"
          >
            <dt
              class="visually-hidden"
            >
              Date created
            </dt>
            <dd>
              12 January 2022 13:14
            </dd>
          </div>
        </dl>
      </article>
    </li>
    <li>
      <article
        class="card"
      >
        <header
          class="card__header"
        >
          <p
            class="card__heading"
          >
            <a
              href="/organisations/1033"
            >
              Org 31
            </a>
          </p>
        </header>
        <dl
          class="card__metadata"
        >
          <div
            class="card__metadatum"
          >
            <dt
              class="visually-hidden"
            >
              Date created
            </dt>
            <dd>
              13 January 2022 13:15
            </dd>
          </div>
        </dl>
      </article>
    </li>
  </ul>
</div>
`;
